src/proteingym/base/msa.py

```diff
@@ -50,16 +90,55 @@ class MSAManifestSection(BaseModel):
     format: MSAFormat = MSAFormat.FASTA
     """The format of the multiple sequence alignment file."""
 
+    weights_path: FilePath | None = Field(default=None, exclude=True)
+    """The weight file for each sequence in the MSA."""
+
+    weights: list[float] | None = None
+    """The weights for each sequence in the MSA (for internal manifest)."""
+
     metadata: dict[str, str] = Field(default_factory=dict)
     """Additional metadata for the multiple sequence alignment."""
 
     @field_validator("path", mode="before", check_fields=True)
     def validate_path(cls, path: Path, info: ValidationInfo) -> Path:
-        """Optionally, extend the path with the `relative_to_path` from the context."""
+        """Optionally, extend the path with the `relative_to_path` from the
+        context."""
+
         if info.context and info.context.get("relative_to_path"):
             path = info.context["relative_to_path"] / path
         return path
 
+    @field_validator("weights_path", mode="before", check_fields=True)
+    def validate_weights_path(
+        cls, weights_path: Path | None, info: ValidationInfo
+    ) -> Path | None:
+        """Extend the weights_path with the `relative_to_path` from the context."""
+
+        if (
+            weights_path is not None
+            and info.context
+            and info.context.get("relative_to_path")
+        ):
+            weights_path = info.context["relative_to_path"] / weights_path
+
+        format = weights_path.suffix[1:].lower()
+        if format not in MSAWeightFormat:
+            raise ValueError(f"Unsupported MSA weight file format: {format}")
+        return weights_path
+
+    @model_validator(mode="after")
+    def check_weights_and_weights_path(
+        self, info: ValidationInfo
+    ) -> "MSAManifestSection":
+        """Ensure that both weights and weights_path are not provided together."""
+
+        if self.weights is not None and self.weights_path is not None:
```

Comment:
If you want to be more Pythonic. Thoug, I am a fan of the explicit `is not None`. Up to you

```suggestion
        if self.weights and self.weights_path:
```